src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/step/rollup/AttemptCreateRollupJobStep.kt

```diff
@@ -0,0 +1,123 @@
+
+    override suspend fun execute(): Step {
+        previousRunRollupId = managedIndexMetaData.actionMetaData?.actionProperties?.rollupId
+        hasPreviousRollupAttemptFailed = managedIndexMetaData.actionMetaData?.actionProperties?.hasRollupFailed
+
+        // Creating a rollup job
+        val rollup = ismRollup.toRollup(indexName)
+        rollupId = rollup.id
+        logger.info("Attempting to create a rollup job $rollupId for index $indexName")
+
+        val indexRollupRequest = IndexRollupRequest(rollup, WriteRequest.RefreshPolicy.IMMEDIATE)
+
+        try {
+            val response = withContext(Dispatchers.IO) { client.execute(IndexRollupAction.INSTANCE, indexRollupRequest).actionGet() }
+            logger.info("Received status ${response.status.status} on trying to create rollup job $rollupId")
+
+            stepStatus = StepStatus.COMPLETED
+            info = mapOf("message" to getSuccessMessage(rollupId!!, indexName))
+        } catch (e: VersionConflictEngineException) {
+            val message = getFailedJobExistsMessage(rollupId!!, indexName)
+            logger.info(message)
+            if (rollupId == previousRunRollupId && hasPreviousRollupAttemptFailed != null && hasPreviousRollupAttemptFailed!!) {
+                withContext(Dispatchers.IO) {
+                    startRollupJob()
+                }
+            } else {
+                stepStatus = StepStatus.COMPLETED
+                info = mapOf("info" to message)
+            }
+        } catch (e: Exception) {
+            val message = getFailedMessage(rollupId!!, indexName)
+            logger.error(message, e)
+            stepStatus = StepStatus.FAILED
+            info = mapOf("message" to message, "cause" to "${e.message}")
+        }
+
+        return this
+    }
+
+    override fun getUpdatedManagedIndexMetaData(currentMetaData: ManagedIndexMetaData): ManagedIndexMetaData {
+        val currentActionMetaData = currentMetaData.actionMetaData
+        return currentMetaData.copy(
+                actionMetaData = currentActionMetaData?.copy(actionProperties = ActionProperties(rollupId = rollupId)),
+                stepMetaData = StepMetaData(name, getStepStartTime().toEpochMilli(), stepStatus),
+                transitionTo = null,
+                info = info
+        )
+    }
+
+    private fun startRollupJob() {
+        logger.info("Attempting to re-start the job $rollupId")
+        try {
+            val startRollupRequest = StartRollupRequest(rollupId!!)
+            client.execute(StartRollupAction.INSTANCE, startRollupRequest).actionGet()
+            stepStatus = StepStatus.COMPLETED
+            info = mapOf("message" to getSuccessRestartMessage(rollupId!!))
+        } catch (e: Exception) {
+            val message = getFailedToStartMessage(rollupId!!)
+            logger.error(message, e)
+            stepStatus = StepStatus.CONDITION_NOT_MET
+            info = mapOf("message" to message)
+        }
+    }
+
+    companion object {
```

Comment:
Can we change the messaging to be consistent with others as there was work to standardize them when they were all over the place. Mainly ending with `[index=$index]` and you can probably remove the `for index` parts.

Reply:
updated